pkg/apis/migration/migration.go

```diff
@@ -0,0 +1,351 @@
+		return "", err
+	}
+	log.Infof("successfully registered volume: %q with CNS. VolumeID: %q", volumePath, volumeID)
+	cnsvSphereVolumeMigration := migrationv1alpha1.CnsVSphereVolumeMigration{
+		ObjectMeta: metav1.ObjectMeta{Name: volumeID},
+		Spec: migrationv1alpha1.CnsVSphereVolumeMigrationSpec{
+			VolumePath: volumePath,
+			VolumeID:   volumeID,
+		},
+	}
+	log.Infof("Saving cnsvSphereVolumeMigration CR: %v", cnsvSphereVolumeMigration)
+	err = volumeMigration.saveVolumeInfo(ctx, &cnsvSphereVolumeMigration)
+	if err != nil {
+		log.Errorf("failed to save cnsvSphereVolumeMigration CR:%v, err: %v", err)
+		return "", err
+	}
+	return volumeID, nil
+}
+
+// GetVolumePath returns VolumePath for given VolumeID
+// Returns an error if not able to retrieve VolumePath.
+func (volumeMigration *volumeMigration) GetVolumePath(ctx context.Context, volumeID string) (string, error) {
+	log := logger.GetLogger(ctx)
+
+	var volumeInfo *migrationv1alpha1.CnsVSphereVolumeMigrationSpec
+	volumeMigration.volumePathToVolumeID.Range(func(volumePath, migrationSpec interface{}) bool {
+		if migrationSpec.(migrationv1alpha1.CnsVSphereVolumeMigrationSpec).VolumeID == volumeID {
+			volumeInfo = migrationSpec.(*migrationv1alpha1.CnsVSphereVolumeMigrationSpec)
+			log.Info("Found volumeInfo %v using volumeID: %q in the cache", volumeInfo, volumeID)
+			return false
+		}
+		return true
+	})
+	volumeMigrationResource := &migrationv1alpha1.CnsVSphereVolumeMigration{}
+	err := volumeMigration.k8sClient.Get(ctx, client.ObjectKey{Name: volumeID}, volumeMigrationResource)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.Errorf("error happened while getting CR for volumeMigration for VolumeID: %q, err: %v", volumeID, err)
+			return "", err
+		}
+	} else {
+		log.Infof("found volume path: %q for VolumeID: %q", volumeMigrationResource.Spec.VolumePath, volumeID)
+		volumeMigration.volumePathToVolumeID.Store(volumeMigrationResource.Spec.VolumePath, volumeID)
+		return volumeMigrationResource.Spec.VolumePath, nil
+	}
+	log.Infof("Could not retrieve mapping of volume path and volume ID in the cache for VolumeID: %q. volume may not be registered", volumeID)
+	volumeIds := []cnstypes.CnsVolumeId{{Id: volumeID}}
+	log.Infof("Calling QueryVolumeInfo using: %v", volumeIds)
+	queryVolumeInfoResult, err := (*volumeMigration.volumeManager).QueryVolumeInfo(ctx, volumeIds)
+	if err != nil {
+		log.Errorf("QueryVolumeInfo failed for volumeID: %s, err: %v", volumeID, err)
+		return "", err
+	}
+	log.Debugf("QueryVolumeInfo successfully returned volumeInfo %v for volumeIDList %v:", spew.Sdump(queryVolumeInfoResult), volumeIds)
+	cnsBlockVolumeInfo := interface{}(queryVolumeInfoResult.VolumeInfo).(*cnstypes.CnsBlockVolumeInfo)
+	fileBackingInfo := interface{}(cnsBlockVolumeInfo.VStorageObject.Config.Backing).(*vim25types.BaseConfigInfoDiskFileBackingInfo)
+	log.Infof("successfully retrieved volume path: %q for volume id: %q", fileBackingInfo.FilePath, volumeID)
```

Comment:
Can we use "VolumeID" or "volume id" in all log messages, so that it will be consistent and easier for debugging while searching specific strings in logs

Reply:
done. used `VolumeID` to be consistent with all other log messages.